source/cdk/cdk_slurm_stack.py

```diff
@@ -4143,102 +4197,29 @@ def create_parallel_cluster_config(self):
 
-        self.parallel_cluster_lambda_layer = aws_lambda.LayerVersion(self, "ParallelClusterLambdaLayer",
-            description = 'ParallelCluster Layer',
-            code = aws_lambda.Code.from_bucket(
-                s3.Bucket.from_bucket_name(self, 'ParallelClusterBucket', f"{self.config['Region']}-aws-parallelcluster"),
-                f"parallelcluster/{self.config['slurm']['ParallelClusterConfig']['Version']}/layers/aws-parallelcluster/lambda-layer.zip"
-            ),
-            compatible_architectures = [
-                aws_lambda.Architecture.ARM_64,
-                aws_lambda.Architecture.X86_64,
-            ],
-            compatible_runtimes = [
-                aws_lambda.Runtime.PYTHON_3_8,
-                aws_lambda.Runtime.PYTHON_3_9,
-                # aws_lambda.Runtime.PYTHON_3_10,
-                # aws_lambda.Runtime.PYTHON_3_11,
-            ],
-        )
-        createParallelClusterAsset = s3_assets.Asset(self, "CreateParallelClusterAsset", path="resources/lambdas/CreateParallelCluster")
-        self.create_parallel_cluster_lambda = aws_lambda.Function(
-            self, "CreateParallelClusterLambda",
-            function_name=f"{self.stack_name}-CreateParallelCluster",
-            description="Create ParallelCluster from config file in s3",
-            memory_size=2048,
-            runtime=aws_lambda.Runtime.PYTHON_3_9,
-            architecture=aws_lambda.Architecture.X86_64,
-            timeout=Duration.minutes(15),
-            log_retention=logs.RetentionDays.INFINITE,
-            handler="CreateParallelCluster.lambda_handler",
-            code=aws_lambda.Code.from_bucket(createParallelClusterAsset.bucket, createParallelClusterAsset.s3_object_key),
-            layers=[self.parallel_cluster_lambda_layer],
-            vpc = self.vpc,
-            allow_all_outbound = True
-        )
-        self.create_parallel_cluster_lambda.add_to_role_policy(
-            statement=iam.PolicyStatement(
-                effect=iam.Effect.ALLOW,
-                actions=[
-                    '*',
-                ],
-                resources=['*']
-                )
-            )
         self.parallel_cluster = CustomResource(
-            self, "CreatedParallelCluster",
+            self, "ParallelCluster",
             service_token = self.create_parallel_cluster_lambda.function_arn,
             properties = {
+                'ParallelClusterConfigJson': json.dumps(self.parallel_cluster_config, sort_keys=False),
+                'ParallelClusterConfigS3Bucket': self.assets_bucket,
+                'ParallelClusterConfigJsonS3Key': self.parallel_cluster_config_json_s3_key,
+                'ParallelClusterConfigYamlS3Key': self.parallel_cluster_config_yaml_s3_key,
                 'Region': self.config['Region'],
                 'ClusterName': self.config['slurm']['ClusterName'],
-                'ParallelClusterConfigJson': self.parallel_cluster_config,
```

Comment:
This was the source of the bug.
This needed to be converted to a json string and then read back into python dict in the lambda.